packages/chat-headless-react/src/ChatHeadlessProvider.tsx

```diff
@@ -25,14 +25,24 @@ export function ChatHeadlessProvider(
   props: ChatHeadlessProviderProps
 ): JSX.Element {
   const { children, config } = props;
-  const headless = useMemo(
-    () => new ChatHeadless(updateClientSdk(config)),
-    [config]
-  );
+  const [deferLoad, setDeferLoad] = useState(config.saveToSessionStorage);
+
+  const headless = useMemo(() => {
+    const configWithoutSession = { ...config, saveToSessionStorage: false };
+    const headless = new ChatHeadless(updateClientSdk(configWithoutSession));
+    return headless;
+  }, [config]);
+  useEffect(() => {
```

Comment:
nit: new line. And can we add a small comment here (or on top of deferLoad) for why this is needed?